xgo.go

```diff
@@ -299,10 +308,32 @@ func compile(image string, config *ConfigFlags, flags *BuildFlags, folder string
 		"-e", fmt.Sprintf("FLAG_BUILDMODE=%s", flags.Mode),
 		"-e", "TARGETS=" + strings.Replace(strings.Join(config.Targets, " "), "*", ".", -1),
 	}
-	for i := 0; i < len(locals); i++ {
-		args = append(args, []string{"-v", fmt.Sprintf("%s:%s:ro", locals[i], mounts[i])}...)
+	if usesModules {
+		args = append(args, []string{"-e", "GO111MODULE=on"}...)
+		args = append(args, []string{"-v", os.Getenv("GOPATH") + ":/go"}...)
```

Comment:
If `GOPATH` is empty, then we should use the default `GOPATH` (`$HOME/go`).  It looks like `go/build.Default.GOPATH` will do the trick.

Also, should probably join using `filepath.ListSeparator` for portability.

Reply:
The base `Dockerfile` currently maps `GOPATH` to `/go`, so all we're doing here is ensuring that we mount the users `GOPATH` this in order to avoid `go build` hitting the network by getting all the packages.

Do need to ensure `GOPATH` is a valid path first, though. That happens in the previous two cases, but not in this new one. I just missed that.

The `"/go"` is actually a `docker`-specific thing, and uses the same cli syntax on Windows.